src/lib/mdns/Resolver_ImplMinimalMdns.cpp

```diff
@@ -35,6 +38,88 @@ namespace chip {
 namespace Mdns {
 namespace {
 
+enum class DiscoveryType
+{
+    kUnknown,
+    kOperational,
+    kCommissionableNode,
+};
+
+class TxtRecordDelegateImpl : public mdns::Minimal::TxtRecordDelegate
+{
+public:
+    TxtRecordDelegateImpl(CommissionableNodeData * nodeData) : mNodeData(nodeData) {}
+    void OnRecord(const mdns::Minimal::BytesRange & name, const mdns::Minimal::BytesRange & value);
+
+private:
+    CommissionableNodeData * mNodeData;
+};
+
+bool IsKey(const mdns::Minimal::BytesRange key, const char * desired)
+{
+    if (key.Size() != strlen(desired))
+    {
+        return false;
+    }
+    return memcmp(key.Start(), desired, key.Size()) == 0;
+}
+
+uint16_t MakeU16(const mdns::Minimal::BytesRange & val)
+{
+    uint32_t u32          = 0;
+    const uint16_t errval = 0xFFFF;
```

Comment:
Errval should be 0, since there is no vendor ID / product ID valid 0 value.

Reply:
done.